benchmarks/distributed/rpc/parameter_server/trainers/DdpTrainer.py

```diff
@@ -1,42 +1,27 @@
 import torch
-import torch.nn as nn
-from torch.nn.parallel import DistributedDataParallel as DDP
 
-from .DdpTrainerBase import DdpTrainerBase
-
-
-class DdpTrainer(DdpTrainerBase):
-
-    class HookState:
-
-        def __init__(self, cref, process_group):
-            r"""
-            A class that holds state information that is needed by the communication hook
-            during the training algorithm.
-            Args:
-                cref (DdpTrainer): reference to the self keyword of the trainer instance
-                process_group (ProcessGroup): distributed process group
-            """
-            self.cref = cref
-            self.process_group = process_group
-            self.batch_number = -1
-
-        def get_key(self, bucket_index):
-            r"""
-            A method that returns an encoded key that represents the current batch and
-            bucket index.
-            Args:
-                bucket_index (int): index of the bucket being processed in backward
-            """
-            return f"{self.batch_number},{bucket_index}"
-
-        def next_batch(self):
-            r"""
-            A method that increments batch_number by 1.
-            """
-            self.batch_number += 1
-
-    def __init__(self, rank, trainer_count, process_group, use_cuda_rpc, server_rref, backend, epochs):
+from .TrainerBase import TrainerBase
+
+
+class DdpTrainer(TrainerBase):
+
+    def __init__(
+        self,
+        rank,
+        trainer_count,
+        process_group,
+        use_cuda_rpc,
+        server_rref,
+        backend, epochs,
+        preprocess_data,
+        create_criterion,
+        create_ddp_model,
+        lr,
+        create_optimizer,
+        HookState,
```

Comment:
Why do we have a class name here? Do you mean sth like `hook_state_class`?

Reply:
Yes I will fix this.